src/asami/query.cljc

```diff
@@ -792,17 +792,17 @@
   "In the :where sequence of query apply f to each EPV pattern."
   [f {:keys [where] :as query}]
   (assoc query
-         :where (map (fn opf [constraint]
-                       (cond
-                         (epv-pattern? constraint)
-                         (f constraint)
+         :where (doall (map (fn opf [constraint]
```

Comment:
`doall` ensures errors are thrown eagerly during the parse step which we test for. The use of `mapv` in the previous commits caused `execute-query` to wrap the constraints in a vector eventually leading to something like

```clj
(asami.common-index/check-for-transitive [?e :is-in ?e2])
```

to occur.